SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchGrid.swift

```diff
@@ -13,9 +13,11 @@ import UIKit
 class TouchGrid {
     // The result of the scanning procedure will be stored in curNode and childNumber variables
     var curNode = Node()                // The node in the tree we are currently scanning
+	var previousCurNode = Node()		//	curNode during the previous scanning step
     var childNumber: Int = 0            // The index of the highlighted child of curNode
```

Comment:
And then maybe you can rename these to `scanChildIndex` and `nextScanChildIndex` if you think it also makes more sense.